crates/typst/src/eval/call.rs

```diff
@@ -293,47 +281,92 @@ pub(crate) fn call_closure(
 /// associated functions would make an addition of a new associated function a breaking
 /// change and prioritizing fields would break associated functions for certain
 /// dictionaries).
-fn eval_callee_args(
-    target: Value,
-    target_span: Span,
-    field: Ident,
-    mut args: Args,
-) -> SourceResult<(Value, Args)> {
-    if let Some(callee) = target.ty().scope().get(&field) {
-        args.insert(0, target_span, target);
-        Ok((callee.clone(), args))
+fn eval_method_call_and_args(
+    target_expr: ast::Expr,
+    method: Ident,
+    args: ast::Args,
+    span: Span,
+    vm: &mut Vm,
+) -> SourceResult<EvaluatedMethod> {
+    // Evaluate the method's target and overall arguments.
+    let (target, mut args) = if is_mutating_method(&method) {
+        // If this looks like a mutable method then we evaluate the arguments before the
+        // target because calling `target_expr.access(vm)` mutably borrows the vm, meaning
+        // we won't be able to evaluate args after due to the borrow checker.
```

Comment:
```suggestion
        // If `method` looks like a mutating method, we evaluate the arguments first,
        // because `target_expr.access(vm)` mutably borrows the `vm`, so that we can't
        // evaluate the arguments after it.
```